var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -552,7 +552,13 @@ def test(self):
             runexe = Executable(join_path(self.prefix,
                                           'lib/petsc/bin/petsc-mpiexec.uni')).command
             runopt = ['-n', '1']
-        w_dir = join_path(self.install_test_root, 'src/ksp/ksp/tutorials')
+        w_dir = join_path(self.test_suite.current_test_cache_dir,
+                          'src', 'ksp', 'ksp', 'tutorials')
+
+        if not os.path.exists(w_dir):
+            print('Skipping petsc test')
```

Comment:
Suggest adding a bit more information on why the test is being skipped ASSUMING this check is preserved/

```suggestion
            print('Skipping petsc test: KSP tutorial example is missing')
```